frontend/src/__tests__/components/home.test.js

```diff
@@ -0,0 +1,129 @@
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import Home from '../../../pages';
+import { useGetBooks } from '../../../api/books';
+import userEvent from '@testing-library/user-event';
+jest.mock('../../../api/books');
+jest.mock('../../../api/authors');
+jest.mock(
+  'next/image',
+  () =>
+    function Image({ src, alt, width, height }) {
+      // eslint-disable-next-line @next/next/no-img-element
+      return (
+        // eslint-disable-next-line
+        <img src={src} alt={alt} style={{ width: width, height: height }} />
+      );
+    }
+);
+
+jest.mock('../../../components/Modal/Modal', () => ({ children }) => {
+  return (
+    <>
+      <div>MODAL</div>
+      {children}
+    </>
+  );
+});
+jest.mock('../../../components/AddBookForm/AddBookForm', () => () => {
+  return <div>Add Book Form</div>;
+});
+jest.mock('../../../components/ViewBookPAge/ViewBookPage', () => () => {
+  return <div>View Book Page</div>;
+});
+
+describe('test of the main index home page', () => {
+  it('should display the default page', () => {
+    const mockUseGetBooks = useGetBooks;
+    mockUseGetBooks.mockReturnValue({
+      bookLoading: false,
+      bookError: false,
+      books: []
+    });
```

Comment:
Duplication of code throughout this test file, how can we simplify this?

Reply:
I think i reduced the usage. I split describe further into has books and no books. Then for the first it i provided the implementation that i wanted (one without books and one with books). I also made the mockUseGetBooks global.